### PR TITLE
certs: add K8sLabels with legacy and set new style labels

### PR DESCRIPTION
**Pull request by kopiczko**

### CODE REVIEW

File: k8s.go [hunk 1]
```diff
@@ -5,17 +5,23 @@ import "fmt"
 // These constants are used when filtering the secrets, to only retrieve the
 // ones we are interested in.
 const (
-	// componentLabel is the label used in the secret to identify a secret
+	// certificateLabel is the label used in the secret to identify a secret
 	// containing the certificate.
-	//
-	// TODO replace with "giantswarm.io/certificate" and add to
-	// https://github.com/giantswarm/fmt.
```
Comment (kopiczko) [inline k8s.go]: See https://github.com/giantswarm/fmt/pull/18

File: k8s.go [hunk 2]
```diff
@@ -53,6 +59,15 @@ var AllCerts = []Cert{
 
 // K8sSecretName returns Kubernetes Secret object name for the certificate name
 // and the guest cluster ID.
-func K8sSecretName(clusterID, certificate Cert) string {
+func K8sSecretName(clusterID string, certificate Cert) string {
 	return fmt.Sprintf("%s-%s", clusterID, certificate)
 }
```
Comment (kopiczko) [inline k8s.go]: I have a feeling this is unused.

File: k8s.go [hunk 3]
```diff
@@ -53,6 +59,15 @@ var AllCerts = []Cert{
...
+
+func K8sSecretLabels(clusterID string, certificate Cert) map[string]string {
+	return map[string]string{
+		certificateLabel:       string(certificate),
+		clusterIDLabel:         clusterID,
+		legacyCertificateLabel: string(certificate),
+		legacyClusterIDLabel:   clusterID,
+	}
+}
```
Comment (kopiczko) [inline k8s.go]: This is meant to be used  by cert-operator and cluster-operator/kubernetesd

File: k8s.go [hunk 4]
```diff
@@ -51,8 +57,19 @@ var AllCerts = []Cert{
 	WorkerCert,
 }
 
-// K8sSecretName returns Kubernetes Secret object name for the certificate name
-// and the guest cluster ID.
-func K8sSecretName(clusterID, certificate Cert) string {
+// K8sName returns Kubernetes object name for the certificate name and
+// the guest cluster ID.
+func K8sName(clusterID string, certificate Cert) string {
```
Comment (kopiczko) [inline k8s.go]: This is also used for CertConfigs thus changed name.

File: k8s.go (hunk 2 above)
Comment (kopiczko) [reply k8s.go]: Actually used now.

**Review by xh3b4sd — APPROVED** [review]
How do we track the legacy transition here? Should we do this more formal as well with an issue?

**Comment by kopiczko** [comment]
No tracking in terms of issue. I just noticed it when doing other things and tried to push it forward a little. I noticed we have the same issue in encryptionkeys.

**Comment by kopiczko** [comment]
https://github.com/giantswarm/giantswarm/issues/3835

**Pull request merged**